spark/src/main/scala/org/apache/spark/delta/sharing/PreSignedUrlCache.scala

```diff
@@ -81,9 +92,14 @@ class CachedTableManager(
         logInfo(s"Updating pre signed urls for $tablePath (expiration time: " +
           s"${new java.util.Date(cachedTable.expiration)})")
         try {
+          val (idToUrl, expiration) = cachedTable.refresher()
           val newTable = new CachedTable(
-            preSignedUrlExpirationMs + System.currentTimeMillis(),
-            cachedTable.refresher(),
+            if (isValidUrlExpirationTime(expiration)) {
+              expiration.min(preSignedUrlExpirationMs + System.currentTimeMillis())
+            } else {
+              preSignedUrlExpirationMs + System.currentTimeMillis()
```

Comment:
If `isValidUrlExpirationTime` returns `false`,  it means `expiration <= (System.currentTimeMillis() + refreshThresholdMs)`. There are two possible cases:
- -1: the server doesn't tell us the expiration time. In this case, we should use `preSignedUrlExpirationMs + System.currentTimeMillis()`, which is the existing behavior.
- The server sends us a pre signed URL whose expiration time that is about to expire. This could be a few possible cases:
  -  clock differences between the server and the client is too large.
  - the client `refreshThresholdMs` is too large.
  - other unknown issues

Is `preSignedUrlExpirationMs + System.currentTimeMillis()` the right value for the second case? Do you want to avoid refreshing URLs every minute for the second case? Anyway, a warning log here would be pretty helpful to debug the second case.

Reply:
yeah, as you described, the reason is not clear, so I'm not sure what's the best behavior, thus I just use what we are currently using. I'm hesitate to let it refresh every minute because the refresh itself anyway takes time (likely longer than 1 min), and may not catch up with `expiration`, so I'd rather let it fail/expire and then debug.

Good idea to add a warning log here, I'll add it with all necessary info.